src/config/config_impl.go

```diff
@@ -129,7 +129,7 @@ func (this *rateLimitDescriptor) loadDescriptors(config RateLimitConfigToLoad, p
 
 		// Value is optional, so the final key for the map is either the key only or key_value.
 		finalKey := descriptorConfig.Key
-		if descriptorConfig.Value != "" {
+		if descriptorConfig.Value != "" && descriptorConfig.Value != finalKey {
```

Comment:
why do we need this?

Reply:
> why do we need this?

I tried to explain in the PR descriptor, but let's have an example to make it more clear. Envoyproxy is passing here  `descriptors` such as:
```
{
  Key: httproute/envoydemo/envoydemo/rule/0/match/0/envoydemo_domain_com
  Value: httproute/envoydemo/envoydemo/rule/0/match/0/envoydemo_domain_com
}
```
Now, `Value` field is optional, but here Envoy Gateway (?) is setting it exactly the same as the `Key` and therefore without that extra check `Value` is appended to `Key` creating a new `finalKey`. IMO, this doesn't make any sense as it's just data duplication and eventually making those Prometheus labels double the length with unnecessary duplicated information. Let's keep them as readable (aka. short) as possible.

Would the code be a bit more understandable, if it would have `&& descriptorConfig.Value != descriptorConfig.Key` instead?

Reply:
yeah, sounds good to me.

Reply:
added some comments as well